### PR TITLE
feat(Styleguide): Docs Page Wrapper for consistent meta information

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

This adds a universal `<Page />` component to all docs pages, in order to make component documentation more informative, consistent, and simple to implement.  Removing the idea of imported docs blocks that need to be added to every story, this new component uses the `parameters` object and automatically adds information to each story page.

For example in `Card.stories.mdx`
```tsx
<Meta
  title="Atoms/Card"
  component={Card}
  parameters={{
    status: 'volatile', // API / Design mutability
    source: 'gamut-labs',  // Package this component is imported from
    figmaId: 'ReGfRNillGABAj5SlITalN', // Figma design doc this page is meant to represent
  }}
  // Other meta arguments
/>
```

_Note_
Adding/accessing this kind of information via the `parameters` object is the best practice as it is a core concept in the storybook ecosystem making integration with other addons and extension to other parts of storybook (Canvas and Navigation) easier.
 
**New Paramaters**
- `figmaId`: Links to the design doc that these stories are based on.  Currently this just adds a link to the header but ostensibly this can be updated [to an addon that will work with canvas ](https://github.com/pocka/storybook-addon-designs)(but may require some changes to how we use figma and storybook).
- `source`:   Adds a link to the npm page of the package that the component or import is imported from.  E.G. `gamut` | `gamut-labs`.
- `status`: Refactors status to automatically be added to every page which defaults to `stable` if the story is specific to a component.  Other pages may add this but we will only see the default if the story is about a component.
- `pageTitle`:  Adds the pages `h1`, this is automatically derived from `title` path if this is not configured.

There are a few other miscellaneous updates in here:
- Removing unused MDX imports (especially unused knobs).
- Small clean up on knobs usage in favor of props table / argTypes. 
- Restructures stories in `Labs` to make it less nested.

* Adds `<Page />` component to consistently format the top head of all story pages.
* Updates story pages to work with new automatically generated formatting.
* Restructures `Labs` section to move all `Experimental` into `Core` as they are closely related making Brand the top-level section.

### Example Screenshot
Screen Shot 2020-12-28 at 12 55 32 PM

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.6.6-alpha.6f6d81.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fea2e32fbdefc134e44eb9b--gamut-preview.netlify.app>https://5fea2e32fbdefc134e44eb9b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fea2e32fbdefc134e44eb9b>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Super!

File: packages/styleguide/.storybook/components/Page/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      subcomponents,
+      status,
+      pageTitle,
+      figmaId,
+      source,
+    },
+  } = useContext(DocsContext);
+
+  const titleString = pageTitle || title.split('/').reverse()[0];
+  const showStatus = Boolean(status) || Boolean(component || subcomponents);
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/styleguide/.storybook/components/Page/index.tsx:53]: Heh refactor relic, nice catch.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.6.6-alpha.523cc6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fecb7162e58a301393f980e--gamut-preview.netlify.app>https://5fecb7162e58a301393f980e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fecb7162e58a301393f980e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.6.7-alpha.1f4bd6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff339537722093fb28bef0c--gamut-preview.netlify.app>https://5ff339537722093fb28bef0c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff339537722093fb28bef0c>Deploy Logs</a></p>

**Pull request merged**